packages/playwright-core/src/server/socksClientCertificatesInterceptor.ts

```diff
@@ -224,21 +224,11 @@ export class ClientCertificatesProxy {
   }
 }
 
-const kClientCertificatesGlobRegex = Symbol('kClientCertificatesGlobRegex');
-
 export function clientCertificatesToTLSOptions(
   clientCertificates: channels.BrowserNewContextOptions['clientCertificates'],
   origin: string
 ): Pick<https.RequestOptions, 'pfx' | 'key' | 'cert'> | undefined {
-  const matchingCerts = clientCertificates?.filter(c => {
-    let regex: RegExp | undefined = (c as any)[kClientCertificatesGlobRegex];
-    if (!regex) {
-      regex = globToRegex(c.origin);
-      (c as any)[kClientCertificatesGlobRegex] = regex;
-    }
-    regex.lastIndex = 0;
-    return regex.test(origin);
-  });
+  const matchingCerts = clientCertificates?.filter(c => c.origin === origin);
```

Comment:
Thought: maybe we should do:

     clientCertificates?.filter(c => new URL(c.origin).origin === origin);

in order to be more user-friendly.

Reply:
Yeah, sounds great, but you'll need to try/catch to avoid an unexpected error.